src/echoclient/EchoClient.php

```diff
@@ -62,51 +66,51 @@ function __construct()
     }
 
     /**
-     * Add an event to echo
+     * Adds a single event to echo
      *
      * @param string $class
      * @param string $source
      * @param array $props
      * @param string|null $userId
+     * @param string|null $timestamp
      * @return bool True if successful, else false
      */
-    public function createEvent($class, $source, array $props = array(), $userId = null)
+    public function createEvent($class, $source, array $props = array(), $userId = null, $timestamp = null)
```

Comment:
Should we subject this method to the same validation for constructing events?

For instance, if we had this method construct an `EchoEvent` instance, it would get the construction-time validation on data types (throwing illegal argument exceptions); if we then constructed an array containing just this element, we'd also get the maximum-event-size-in-MB validation for free.

My suggestion would be to make this method construct an `EchoEvent` and then `return $this->createEvents([$echoEvent]);` accordingly.

Reply:
done